actor/manager/manager.go

```diff
@@ -0,0 +1,240 @@
+	if err != nil {
+		return nil, actorErr.ErrActorInvokeFailed
+	}
+	m.getAndCreateActorContainerIfNotExist(actorID).GetActor().SaveState()
+	return rspData, actorErr.Success
+}
+
+// DetectiveActor removes actor from actor manager
+func (m *DefaultActorManager) DetectiveActor(actorID string) actorErr.ActorErr {
+	_, ok := m.activeActors.Load(actorID)
+	if !ok {
+		return actorErr.ErrActorIDNotFound
+	}
+	m.activeActors.Delete(actorID)
+	return actorErr.Success
+}
+
+// InvokeReminder invoke reminder function with given params
+func (m *DefaultActorManager) InvokeReminder(actorID, reminderName string, params []byte) actorErr.ActorErr {
+	if m.factory == nil {
+		return actorErr.ErrActorFactoryNotSet
+	}
+	reminderParams := &api.ActorReminderParams{}
+	if err := json.Unmarshal(params, reminderParams); err != nil {
+		fmt.Println("unmarshal reminder param error = ", err)
+		return actorErr.ErrRemindersParamsInvalid
+	}
+	targetActor, ok := m.getAndCreateActorContainerIfNotExist(actorID).GetActor().(actor.ReminderCallee)
+	if !ok {
+		return actorErr.ErrReminderFuncUndefined
+	}
+	targetActor.ReminderCall(reminderName, reminderParams.Data, reminderParams.DueTime, reminderParams.Period)
+	return actorErr.Success
+}
+
+// InvokeTimer invoke timer callback function with given  params
+func (m *DefaultActorManager) InvokeTimer(actorID, timerName string, params []byte) actorErr.ActorErr {
+	if m.factory == nil {
+		return actorErr.ErrActorFactoryNotSet
+	}
+	timerParams := &api.ActorTimerParam{}
+	if err := json.Unmarshal(params, timerParams); err != nil {
+		fmt.Println("unmarshal reminder param error = ", err)
+		return actorErr.ErrTimerParamsInvalid
+	}
+	_, aerr := m.getAndCreateActorContainerIfNotExist(actorID).Invoke(timerParams.CallBack, timerParams.Data)
+	return aerr
+}
+
+func getAbsctractMethodMap(rcvr interface{}) map[string]*MethodType {
+	s := new(Service)
+	s.rcvrType = reflect.TypeOf(rcvr)
+	s.rcvr = reflect.ValueOf(rcvr)
+	sname := reflect.Indirect(s.rcvr).Type().Name()
+	if !isExported(sname) {
+		s := "type " + sname + " is not exported"
+		panic(s)
+	}
+	return suitableMethods(s.rcvrType)
+}
+
+func isExported(name string) bool {
+	s, _ := utf8.DecodeRuneInString(name)
+	return unicode.IsUpper(s)
+}
+
+// Service is description of service
+type Service struct {
+	rcvr     reflect.Value
+	rcvrType reflect.Type
+}
+
+// MethodType is description of service method.
+type MethodType struct {
+	method    reflect.Method
+	ctxType   reflect.Type   // request context
+	argsType  []reflect.Type // args except ctx, include replyType if existing
+	replyType reflect.Type   // return value, otherwise it is nil
+}
+
+// suitableMethods returns suitable Rpc methods of typ
+func suitableMethods(typ reflect.Type) map[string]*MethodType {
+	methods := make(map[string]*MethodType)
+	for m := 0; m < typ.NumMethod(); m++ {
+		method := typ.Method(m)
+		if mt, err := suiteMethod(method); mt != nil && err == nil {
```

Comment:
Is it designed to skip error?

Reply:
Yes, for illegal methods, we do not export.
I thought I could add a log for illegal method to mention here.